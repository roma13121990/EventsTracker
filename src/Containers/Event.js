import React from 'react';
import FaMapMarker from 'react-icons/lib/fa/map-marker'

export default class Event extends React.Component {

    getTime = () => {
        let hours = new Date(this.props.event.datetime).getHours();
        let min = new Date(this.props.event.datetime).getMinutes();
        min = min === 0 ? "00" : min;
        return hours+ ":" +min
    };

    getDate = () => {
        return new Date(this.props.event.datetime).getDate();
    };

    getMonth = () => {
        let monthList = ["JAN",'FEB','MAR','APR','MAY','JUN','JUL','AUG','SEP','OCT','NOV','DEC']
        let month = new Date(this.props.event.datetime).getMonth();
        return monthList[month]
    };

    render() {
        let {venue, url} = this.props.event;
        let location = `https://www.google.com/maps?q=${venue.latitude},${venue.longitude}`
        return (
            <li className="ev-card-container col-sm-12 col-xs-12 col-md-6 col-lg-4">
                <section className="evt-card">
                    <div className="evt-date-col">
                        <p>{this.getDate()}</p>
                        <p>{this.getMonth()}</p>
                        <p>{this.getTime()}</p>
                        <p>Onwards</p>

                    </div>
                    <div className="evt-info-col">
                        <p title={venue.name}>{venue.name}</p>
                        <p title={venue.city+", "+venue.country}>{venue.city}, {venue.country}</p>
                        <a href={location} target="_blank"><FaMapMarker />Show on map</a>

                    </div>
                    <footer>
                        <a className="Book-now" href={url} target="_blank">Tickets & RSVP</a>
                    </footer>

                </section>
            </li>
        )
    }
}