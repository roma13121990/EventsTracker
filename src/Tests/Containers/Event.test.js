import React from 'react'
import renderer from 'react-test-renderer'
import Event from "../../Containers/Event";

describe('<Event />', () => {
    const propsWithMinutes = {
        "event":
            {
                "offers": [{
                    "type": "Tickets",
                    "url": "https://www.bandsintown.com/t/1001879056?app_id=ID&came_from=267",
                    "status": "available"
                }],
                "venue": {
                    "name": "Madison Square Garden",
                    "country": "United States",
                    "region": "NY",
                    "city": "New York",
                    "latitude": "40.7507387",
                    "longitude": "-73.9937978"
                },
                "datetime": "2018-10-15T19:30:09",
                "on_sale_datetime": "",
                "description": "",
                "lineup": ["Maroon 5", "Julia Michaels"],
                "id": "1001879056",
                "artist_id": "510",
                "url": "https://www.bandsintown.com/e/1001879056?app_id=ID&came_from=267"
            }
    };

    const propsWithoutMinutes = {
        "event":
            {
                "offers": [{
                    "type": "Tickets",
                    "url": "https://www.bandsintown.com/t/1008981694?app_id=ID&came_from=267",
                    "status": "available"
                }],
                "venue": {
                    "name": "Parque Fundidora",
                    "country": "Mexico",
                    "region": "NL",
                    "city": "Monterrey",
                    "latitude": "25.6666667",
                    "longitude": "-100.3166667"
                },
                "datetime": "2018-08-25T19:00:00",
                "on_sale_datetime": "2018-05-16T21:30:00",
                "description": "",
                "lineup": ["Maroon 5", "Alvaro DÃ­az", "Gareth Emery", "Jack Novak", "NERVO", "Kali Uchis", "French Montana", "Grouplove", "GG Magree"],
                "id": "1008981694",
                "artist_id": "510",
                "url": "https://www.bandsintown.com/e/1008981694?app_id=ID&came_from=267"
            }
    };
    it('should generate snapshot for Event component', () => {
        const tree = renderer.create(<Event {...propsWithMinutes}/>).toJSON();
        expect(tree).toMatchSnapshot()
    })

    it('should generate snapshot for Event component', () => {
        const tree = renderer.create(<Event {...propsWithoutMinutes}/>).toJSON();
        expect(tree).toMatchSnapshot()
    })


});