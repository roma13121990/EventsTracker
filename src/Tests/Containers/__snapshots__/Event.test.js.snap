// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Event /> should generate snapshot for Event component 1`] = `
<li
  className="ev-card-container col-sm-12 col-xs-12 col-md-6 col-lg-4"
>
  <section
    className="evt-card"
  >
    <div
      className="evt-date-col"
    >
      <p>
        16
      </p>
      <p>
        OCT
      </p>
      <p>
        1:00
      </p>
      <p>
        Onwards
      </p>
    </div>
    <div
      className="evt-info-col"
    >
      <p
        title="Madison Square Garden"
      >
        Madison Square Garden
      </p>
      <p
        title="New York, United States"
      >
        New York
        , 
        United States
      </p>
      <a
        href="https://www.google.com/maps?q=40.7507387,-73.9937978"
        target="_blank"
      >
        <svg
          fill="currentColor"
          height="1em"
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "color": undefined,
              "verticalAlign": "middle",
            }
          }
          viewBox="0 0 40 40"
          width="1em"
        >
          <g>
            <path
              d="m25.6 14.3q0-2.4-1.6-4.1t-4.1-1.6-4 1.6-1.7 4.1 1.7 4 4 1.7 4.1-1.7 1.6-4z m5.8 0q0 2.4-0.8 4l-8.1 17.3q-0.4 0.7-1.1 1.1t-1.5 0.4-1.5-0.4-1-1.1l-8.2-17.3q-0.7-1.6-0.7-4 0-4.7 3.3-8.1t8.1-3.3 8.1 3.3 3.4 8.1z"
            />
          </g>
        </svg>
        Show on map
      </a>
    </div>
    <footer>
      <a
        className="Book-now"
        href="https://www.bandsintown.com/e/1001879056?app_id=ID&came_from=267"
        target="_blank"
      >
        Tickets & RSVP
      </a>
    </footer>
  </section>
</li>
`;

exports[`<Event /> should generate snapshot for Event component 2`] = `
<li
  className="ev-card-container col-sm-12 col-xs-12 col-md-6 col-lg-4"
>
  <section
    className="evt-card"
  >
    <div
      className="evt-date-col"
    >
      <p>
        26
      </p>
      <p>
        AUG
      </p>
      <p>
        0:30
      </p>
      <p>
        Onwards
      </p>
    </div>
    <div
      className="evt-info-col"
    >
      <p
        title="Parque Fundidora"
      >
        Parque Fundidora
      </p>
      <p
        title="Monterrey, Mexico"
      >
        Monterrey
        , 
        Mexico
      </p>
      <a
        href="https://www.google.com/maps?q=25.6666667,-100.3166667"
        target="_blank"
      >
        <svg
          fill="currentColor"
          height="1em"
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "color": undefined,
              "verticalAlign": "middle",
            }
          }
          viewBox="0 0 40 40"
          width="1em"
        >
          <g>
            <path
              d="m25.6 14.3q0-2.4-1.6-4.1t-4.1-1.6-4 1.6-1.7 4.1 1.7 4 4 1.7 4.1-1.7 1.6-4z m5.8 0q0 2.4-0.8 4l-8.1 17.3q-0.4 0.7-1.1 1.1t-1.5 0.4-1.5-0.4-1-1.1l-8.2-17.3q-0.7-1.6-0.7-4 0-4.7 3.3-8.1t8.1-3.3 8.1 3.3 3.4 8.1z"
            />
          </g>
        </svg>
        Show on map
      </a>
    </div>
    <footer>
      <a
        className="Book-now"
        href="https://www.bandsintown.com/e/1008981694?app_id=ID&came_from=267"
        target="_blank"
      >
        Tickets & RSVP
      </a>
    </footer>
  </section>
</li>
`;
